### PR TITLE
Update nanomsg types for version 4.2

### PR DESCRIPTION
**Pull request by toebeann**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/nickdesaulniers/node-nanomsg
- [X] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by toebeann** [comment]
@titan Is there anything I can do to help get this reviewed?

My main interest is in having access to the proper EventEmitter interface which nanomsg `Socket` class implement, as well as access to the `closed` property is also useful for working with multiple sockets as independent workers on the same endpoint.

Obviously, I can use ugly `<any>` casts etc. to get the job done, but I'd rather use the correct typings.

I've been using my own custom version of the typings based on yours for a while now in a local testing environment for quite a while and have not run into any issues.

File: types/nanomsg/index.d.ts [hunk 1]
```diff
@@ -1,36 +1,148 @@
...
+// Definitions by: Tobey Blaber <https://github.com/toebeann>
+//                 Titan <https://github.com/titan>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
 /// <reference types="node" />
 
+import { EventEmitter } from "events";
+
 export interface Options {
-  raw?: boolean | undefined;
```
Comment (andrewbranch) [inline types/nanomsg/index.d.ts:9]: These were written like this on purpose—`raw?: boolean` and `raw?: boolean | undefined` mean different things in `--exactOptionalPropertyTypes`.

**Review by andrewbranch — CHANGES_REQUESTED** [review]

File: types/nanomsg/index.d.ts (hunk 1 above)
Comment (toebeann) [reply types/nanomsg/index.d.ts:9]: I see, I'm not familiar with that flag. Thought it was just redundant semantics, my bad - will fix up :)

Comment (toebeann) [reply types/nanomsg/index.d.ts:9]: Resolved by [bab25b0](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/61223/commits/bab25b0d4468e81d61889199b0bdef01786e78a1)

**Review by andrewbranch — APPROVED** [review]
Thanks!

**Comment by toebeann** [comment]
Ready to merge 💜

**Pull request merged**